lstchain/image/modifier.py

```diff
@@ -53,33 +56,59 @@ def add_noise_in_pixels(rng, image, extra_noise_in_dim_pixels,
 
     return image
 
+@njit(cache=True)
+def set_numba_seed(seed):
+    np.random.seed(seed)
 
-def smear_light_in_pixels(image, camera_geom, smeared_light_fraction):
+@njit(cache=True)
+def random_psf_smearer(image, fraction, indices, indptr):
     """
-
     Parameters
     ----------
     image: charges (p.e.) in the camera
 
-    camera_geom: camera geometry
+    indices: camera_geometry.neighbor_matrix_sparse.indices
+
+    indptr: camera_geometry.neighbor_matrix_sparse.indptr
 
-    smeared_light_fraction: fraction of the light in a pixel that will be
-    distributed equally among its immediate surroundings, i.e. immediate
-    neighboring pixels. Some light is lost for pixels which are at the
-    camera edge and hence don't have all possible neighbors
+    fraction: fraction of the light in a pixel that will be
+    distributed among its immediate surroundings, i.e. immediate
+    neighboring pixels, according to Poisson statistics. Some light is
+    lost for pixels which are at the camera edge and hence don't have all
+    possible neighbors
 
     Returns
     -------
     Modified (smeared) image
 
     """
 
-    # Move a fraction of the light in each pixel (fraction) into its neighbors,
-    # to simulate a worse PSF:
-    q_smeared = (image * camera_geom.neighbor_matrix *
-                 smeared_light_fraction / N_PIXEL_NEIGHBORS)
-    # Light remaining in pixel:
-    q_remaining = image * (1 - smeared_light_fraction)
-    image = q_remaining + np.sum(q_smeared, axis=1)
+    new_image = image.copy()
 
-    return image
+    for pixel in range(len(image)):
+
+        if image[pixel] <= 0:
+            continue
+
+        to_smear = np.random.poisson(image[pixel] * fraction)
+
+        if to_smear == 0:
```

Comment:
how can this be exactly 0 if `image[pixel]` is not 0?

Reply:
because `to_smear` is drawn from a poissonian distribution